protocol/activity_center_persistence_test.go

```diff
@@ -205,17 +205,22 @@ func (s *ActivityCenterPersistenceTestSuite) Test_DeleteActivityCenterNotificati
 	}
 
 	// Test: soft delete the notifications that have Message.ID == messages[1].ID
-	// or LastMessage.ID == chat.LastMessage.
 	_, err = p.DeleteActivityCenterNotificationForMessage(chat.ID, messages[1].ID, currentMilliseconds())
 	s.Require().NoError(err)
 
-	for _, id := range []types.HexBytes{nID2, nID3} {
-		notif, err = p.GetActivityCenterNotificationByID(id)
-		s.Require().NoError(err, notif.ID)
-		s.Require().True(notif.Deleted, notif.ID)
-		s.Require().True(notif.Dismissed, notif.ID)
-		s.Require().True(notif.Read, notif.ID)
-	}
+	notif, err = p.GetActivityCenterNotificationByID(nID2)
+	s.Require().NoError(err, notif.ID)
+	s.Require().True(notif.Deleted, notif.ID)
+	s.Require().True(notif.Dismissed, notif.ID)
+	s.Require().True(notif.Read, notif.ID)
+
+	// Check that notifications with LastMessage.ID == messages[1].ID will remain.
+	// Unless the notification is of type NewOneToOne or NewPrivateGroupChat.
+	notif, err = p.GetActivityCenterNotificationByID(nID3)
+	s.Require().NoError(err)
+	s.Require().False(notif.Deleted, notif.ID)
+	s.Require().False(notif.Dismissed, notif.ID)
+	s.Require().False(notif.Read, notif.ID)
```

Comment:
Here we're updating the original test case for deleting a message to not expect that we should notifications when they reference the deleted message via the `LastMessage` field.